src/corecel/math/ArrayUtils.hh

```diff
@@ -149,6 +155,28 @@ CELER_FUNCTION Array<T, N> make_unit_vector(Array<T, N> const& v)
     return result;
 }
 
+//---------------------------------------------------------------------------//
+/*!
+ * Return the component of \em x that is orthogonal to the unit vector \em y.
+ *
+ * In this implementation, \em y must be normalized, and the result is normal.
+ *
+ * \f[
+  \mathbf{x}' \gets
+ \mathbf{x} - \frac{\mathbf{x} \cdot \mathbf{y}}{\mathbf{y} \cdot \mathbf{y}}
+\mathbf{y}
```

Comment:
Maybe just to be explicit that this won't normalize y:
```suggestion
  \mathbf{x}' \gets \mathbf{x} - (\mathbf{x} \cdot \mathbf{y}) \mathbf{y}, \quad \|\mathbf{y}\| = 1
```

Reply:
Oh yep indeed.